src/baseTrie.js

```diff
@@ -24,752 +22,759 @@ module.exports = Trie
- * @memberof Trie
- * @param {Buffer} key
- * @param {Function} callback A callback `Function`, which is given the arguments `err` - for errors that may have occured and `value` - the found value in a `Buffer` or if no value was found `null`.
- */
-Trie.prototype.getRaw = function (key, cb) {
-  key = ethUtil.toBuffer(key)
-
-  function dbGet (db, cb2) {
-    db.get(key, {
-      keyEncoding: 'binary',
-      valueEncoding: 'binary'
-    }, function (err, foundNode) {
-      if (err || !foundNode) {
-        cb2(null, null)
-      } else {
-        cb2(null, foundNode)
-      }
+      cb(err, value)
     })
   }
-  asyncFirstSeries(this._getDBs, dbGet, cb)
-}
 
-// retrieves a node from dbs by hash
-Trie.prototype._lookupNode = function (node, cb) {
-  if (TrieNode.isRawNode(node)) {
-    cb(new TrieNode(node))
-  } else {
-    this.getRaw(node, function (err, value) {
-      if (err) {
-        throw err
-      }
-
-      if (value) {
-        value = new TrieNode(rlp.decode(value))
-      }
+  /**
+   * Stores a given `value` at the given `key`
+   * @method put
+   * @memberof Trie
+   * @param {Buffer|String} key
+   * @param {Buffer|String} Value
+   * @param {Function} cb A callback `Function` which is given the argument `err` - for errors that may have occured
+   */
+  put (key, value, cb) {
+    key = ethUtil.toBuffer(key)
+    value = ethUtil.toBuffer(value)
+
+    if (!value || value.toString() === '') {
+      this.del(key, cb)
+    } else {
+      cb = callTogether(cb, this.sem.leave)
+
+      this.sem.take(() => {
+        if (this.root.toString('hex') !== ethUtil.SHA3_RLP.toString('hex')) {
+          // first try to find the give key or its nearst node
+          this.findPath(key, (err, foundValue, keyRemainder, stack) => {
+            if (err) {
+              return cb(err)
+            }
+            // then update
+            this._updateNode(key, value, keyRemainder, stack, cb)
+          })
+        } else {
+          this._createInitialNode(key, value, cb) // if no root initialize this trie
+        }
+      })
+    }
+  }
 
-      cb(value)
+  /**
+   * deletes a value given a `key`
+   * @method del
+   * @memberof Trie
+   * @param {Buffer|String} key
+   * @param {Function} callback the callback `Function`
+   */
+  del (key, cb) {
+    key = ethUtil.toBuffer(key)
+    cb = callTogether(cb, this.sem.leave)
+
+    this.sem.take(() => {
+      this.findPath(key, (err, foundValue, keyRemainder, stack) => {
+        if (err) {
+          return cb(err)
+        }
+        if (foundValue) {
+          this._deleteNode(key, stack, cb)
+        } else {
+          cb()
+        }
+      })
     })
   }
-}
 
-// TODO: remove the proxy method when changing the caching
-Trie.prototype._putRaw = function (key, val, cb) {
-  function dbPut (db, cb2) {
-    db.put(key, val, {
-      keyEncoding: 'binary',
-      valueEncoding: 'binary'
-    }, cb2)
+  /**
+   * Retrieves a raw value in the underlying db
+   * @method getRaw
```

Comment:
This is missing the ``@memberof`` property and it is therefor not included in the docs (unintended I suppose?).